test/integration/server-discovery-and-monitoring/server_discover_and_monitoring.test.ts

```diff
@@ -8,3 +15,112 @@ describe('SDAM Unified Tests (Node Driver)', function () {
   );
   runUnifiedSuite(clonedAndAlteredSpecTests);
 });
+
+describe('Monitoring rtt tests', function () {
+  let client: MongoClient;
+  let heartbeatDurations: Record<string, number[]>;
+  const HEARTBEATS_TO_COLLECT = 200; // Wait for 200 total heartbeats. This is high enough to work for standalone, sharded and our typical 3-node replica set topology tests
+  const IGNORE_SIZE = 20;
+  const DELAY_MS = 10;
+  let count: number;
+  const ee = new EventEmitter();
+
+  const listener = (ev: ServerHeartbeatSucceededEvent) => {
+    if (!client.topology.s.servers.has(ev.connectionId)) return;
+    count++;
+    if (count < IGNORE_SIZE) {
+      return;
+    }
+
+    heartbeatDurations[ev.connectionId].push(ev.duration);
+
```

Comment:
```suggestion
    (heartbeatDurations[ev.connectionId] ??= []).push(ev.duration);
```

Not trying to code golf you here, but then you don't need to "setup" heartbeatDurations. This combined with removing the !has check may change the output, so I'm curious to first understand what connectionIds aren't in the server map.